python/tank/descriptor/descriptor_bundle.py

```diff
@@ -392,13 +392,17 @@ class EngineDescriptor(BundleDescriptor):
     Descriptor that describes a Toolkit Engine
     """
 
-    def __init__(self, sg_connection, io_descriptor):
+    def __init__(self, sg_connection, io_descriptor, bundle_cache_root_override, fallback_roots):
```

Comment:
Descriptor classes had to have their CTOR params unified in order to implement the new abstract factory pattern in the simplest way.